File: src/main/java/net/mreunionlabs/tapgradle/generate/generator/ManifestGenerator.java
package net.mreunionlabs.tapgradle.generate.generator;

import net.mreunionlabs.tapgradle.generate.GeneratePluginExtension;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.PrintWriter;

/**
 * Created by abangkis on 3/19/2016.
 */
public class ManifestGenerator {

    private static final String MANIFEST = "MANIFEST.MF";

    public void createFile(GeneratePluginExtension ext) {
        File assetFolder = ext.getAssetFolder();
        if (assetFolder != null) {

            File manifest = new File(assetFolder, MANIFEST);

            try (PrintWriter writer = new PrintWriter(manifest)) {
                writer.println("Manifest-Version: 1.0");
                writer.println("Class-Path: ");
                writer.close();
            } catch (FileNotFoundException e) {
                e.printStackTrace();
            }
        }
        else {
            System.err.println("MANIFEST GEN: asset folder is null");
        }
    }
}

File: src/main/kotlin/net/mreunionlabs/tapgradle/generate/generator/ManifestGenerator.kt
package net.mreunionlabs.tapgradle.generate.generator

import net.mreunionlabs.tapgradle.generate.GeneratePluginExtension
import java.io.File
import java.io.FileNotFoundException
import java.io.PrintWriter

class ManifestGenerator {

    fun createFile(ext: GeneratePluginExtension) {
        val assetFolder = ext.assetFolder
        if (assetFolder != null) {

            val manifest = File(assetFolder, MANIFEST)

            try {
                PrintWriter(manifest).use { writer ->
                    writer.println("Manifest-Version: 1.0")
                    writer.println("Class-Path: ")
                    writer.close()
                }
            } catch (e: FileNotFoundException) {
                e.printStackTrace()
            }

        } else {
            System.err.println("MANIFEST GEN: asset folder is null")
        }
    }

    companion object {

        private val MANIFEST = "MANIFEST.MF"
    }
}
